animation_nodes/data_structures/attributes/attribute.pyx

```diff
@@ -0,0 +1,120 @@
+from .. lists.base_lists cimport (
+    LongList,
+    FloatList,
+    ColorList,
+    BooleanList,
+    Vector2DList,
+    Vector3DList,
+)
+
+cdef class Attribute:
+    def __cinit__(self, str name = None,
+                        AttributeType type = CUSTOM,
+                        AttributeDomain domain = POINT,
+                        AttributeDataType dataType = FLOAT,
+                        object data = None):
+
+        self.name = name
+        self.type = type
+        self.domain = domain
+        self.dataType = dataType
+        self.data = data
+
+    def copy(self):
+        return Attribute(self.name, self.type, self.domain, self.dataType,
+                         self.data.copy())
+
+    def repeated(self, Py_ssize_t amount = -1):
+        return Attribute(self.name, self.type, self.domain, self.dataType,
+                         self.data.repeated(amount = amount))
+
+    def extend(self, object data):
+        self.data.extend(data)
+
+    @property
```

Comment:
I don't like class properties to be honest. I would use standard methods for this.
Also, most of those asString properties are not needed now that we can access the enums in Python. So those should be removed.

Reply:
Removed.